twopaneview/examples/TwoPaneView.js

```diff
@@ -0,0 +1,215 @@
+    spanning: DualScreenInfo.isSpanning, 
+    panePriority: this.props.panePriority,
+    panePriorityVerticalSpanning: this.props.panePriorityVerticalSpanning,
+    paneMode: this.props.paneMode ? this.props.paneMode: PaneMode.Auto,
+    twoPaneWidth: this.props.twoPaneWidth ? this.props.twoPaneWidth: 640,
+    onModeChanged: this.props.onModeChanged ? this.props.onModeChanged : ()=>{},
+  };
+
+  componentDidMount() {
+    Dimensions.addEventListener('change', this._handleDimensionsChange);
+  }
+
+  componentWillUnmount() {
+    Dimensions.removeEventListener('change', this._handleDimensionsChange);
+  }
+
+  _handleDimensionsChange = dimensions => {
+    this.setState({
+      dims: dimensions.window,
+      spanning: DualScreenInfo.isSpanning 
+    });
+    this.state.onModeChanged();
+  };
+
+  render() {    
+
+    let direction ='row';
+
+    return (
+      <View style={{flexDirection: direction, width: this.state.dims.width, height:this.state.dims.height}}>
+        {this.renderChildPanes()}
+      </View>
+    );
+  }
+
+  renderChildPanes() {
+    const children = React.Children.toArray(this.props.children);
+
+    if (this.state.spanning) {
+      if (this.state.paneMode === PaneMode.Single || this.state.dims.height > this.state.dims.width  && this.state.panePriorityVerticalSpanning) {
+        if (this.state.panePriorityVerticalSpanning === TPV_PanePriority.Pane1) {
+          return this.renderPane1(this.getEntireSize());
+        }
+        else {
+          return this.renderPane2(this.getEntireSize());
+        }
+      }
+      return this.renderBothPanes();
+    }
+    if (this.state.paneMode === PaneMode.Dual) {
+      return this.renderDualPanes();
+    }
+    if (this.state.panePriority === TPV_PanePriority.Pane1) {
+      return this.renderPane1(this.getEntireSize());
+    }
+    return this.renderPane2(this.getEntireSize());
+  }
+
+  renderDualPanes() {
+    const children = React.Children.toArray(this.props.children);
+
+    const items = [];
+    if (children.length > 0) {
+      items.push(this.renderPane1(this.getLeftSize()));
+    }
+
+
+    if (children.length > 1) {
+      items.push(this.renderPane2(this.getRightSize()));
+    }
+
+    return items;
+  }
+
+  renderBothPanes() {
+    let horizontal = this.state.dims.width >= this.state.dims.height;
+    const children = React.Children.toArray(this.props.children);
+
+    const items = [];
+    if (children.length > 0) {
+      items.push(this.renderPane1(horizontal ? this.getLeftSize() : this.getTopSize()));
+    }
+
+    items.push(this.renderSeparator());
+
+    if (children.length > 1) {
+      items.push(this.renderPane2(horizontal ? this.getRightSize() : this.getBottomSize()));
+    }
+
+    return items;
+  }
+
+  renderPane1(size) {
+    const children = React.Children.toArray(this.props.children);
+    if (children.length > 0) {
+      return (
+        <View key="pane1" style={{width:size.width, height:size.height}}>
```

Comment:
why would you use  width and height and Dimensions in General?

I believe the only way is either to use flexbox and do  whole sizing natively. @kmelmon can you comment on this?

Reply:
This is somewhat complex to get right.  Will take this offline to discuss.

---
In reply to: [391525771](https://github.com/microsoft/react-native-dualscreen/pull/5#discussion_r391525771) [](ancestors = 391525771)

Reply:
After discussing, I think all that's needed is to set flex:1 for both pane1 and pane2.

---
In reply to: [392512149](https://github.com/microsoft/react-native-dualscreen/pull/5#discussion_r392512149) [](ancestors = 392512149,391525771)